src/lib/dialog/dialog.ts

```diff
@@ -91,11 +91,19 @@ export class MdDialog {
    */
   open<T>(componentOrTemplateRef: ComponentType<T> | TemplateRef<T>,
           config?: MdDialogConfig): MdDialogRef<T> {
+
+    const inProgressDialog = this._openDialogs.find(dialog => dialog._isAnimating());
```

Comment:
I see how that would be useful, but it's probably too much trouble. There's only a 400ms window after you open a dialog where this can happen.